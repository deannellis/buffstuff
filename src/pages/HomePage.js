import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter, Link} from "react-router-dom";
import moment from 'moment';

import { startWorkout } from '../actions/inProgressWorkout';
import { setTargetMacros, updateMacro, setCurrentDate, resetCurrent } from '../actions/macros';
import Dashboard from '../components/Dashboard';
import Button from '../components/Button';
import SideNav from '../components/SideNav';

class HomePage extends Component {
    constructor(props) {
        super(props);
        this.state = {
            today: moment()
        }
    }

    onStartWorkout = () => {
        this.props.dispatch(startWorkout(this.props.liftVariant));
        this.props.history.push('/workout');
    }

    onSetMacros = macros => { this.props.dispatch(setTargetMacros(macros)) }

    onUpdateMacro = update => {
        this.props.dispatch(updateMacro(update))
    }

    componentDidMount() {
        const { dateObject } = this.props.macros;
        const storedDate = dateObject._d
        const { today } = this.state;
        if(dateObject === null || dateObject === 0) {
            console.log('SETTING DATE DUE TO NULL OR 0 VALUE');
            this.props.dispatch(setCurrentDate({ currentDate: today }));
        }else if(!(moment().isSame(storedDate, 'day'))) {
            console.log('RESETING CURRENT MACROS BASED ON OUTDATED DATE');
            this.props.dispatch(resetCurrent());
            this.props.dispatch(setCurrentDate({ currentDate: today }));
        }
    }

    render() { 
        if(Object.entries(this.props.masterWeights).length === 0 && this.props.masterWeights.constructor === Object) {
            return (
                <>
                    <h2>Welcome to flext!</h2>
                    <p>Let's begin by entering your starting weights</p>
                    <Link to="/onboarding">
                        <Button variant="primary">
                            Enter Weights
                        </Button>
                    </Link>
                </>
            );
        }
        return (
            <div className="page--with-side-nav">
                <SideNav path={this.props.match.path} />
                <Dashboard
                    liftVariant={this.props.liftVariant}
                    masterWeights={this.props.masterWeights} 
                    workouts={this.props.workouts}
                    onStartWorkout={this.onStartWorkout}
                    macros={this.props.macros}
                    onSetMacros={this.onSetMacros}
                    onUpdateMacro={this.onUpdateMacro}
                />
            </div>
        );
    }
}

const mapStateToProps = state => {
    return {
        masterWeights: state.masterWeights,
        liftVariant: state.liftVariant,
        workouts: state.workouts,
        macros: state.macros,
    };
}
 
export default withRouter(connect(mapStateToProps)(HomePage));