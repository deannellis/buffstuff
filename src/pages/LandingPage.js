import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

import { startLogin } from '../actions/auth';
import Button from '../components/Button';
import AppFooter from '../components/AppFooter';

const LoginPage = ({ startLogin }) => {
	return (
		<div className="landing-page">
			<div className="landing-page__hero">
				<div>
					<h1>flext takes the guess work out of weight lifting</h1>
					<Button clickHandler={startLogin()}>
						<img
							src="./images/64px-Google__G__Logo.webp"
							alt="Google mark"
							height="32px"
						/>
						Log in with Google
					</Button>
				</div>
				<div className="landing-page__hero-image">
					<img
						src="./images/landingIllustration.png"
						alt="illustration of weightlifter"
					/>
				</div>
			</div>
			<div className="landing-page__section-2">
				<div>
					<img
						src="./images/workoutComponents.png"
						alt="Components from within the application"
					/>
				</div>
				<div>
					<h1>
						flext knows what your next workout looks like so you don&apos;t have
						to
					</h1>
				</div>
			</div>
			<div className="landing-page__section-3">
				<div>
					<h1>See your gains increase as you progress</h1>
				</div>
				<div>
					<img
						src="./images/weightGraph.png"
						alt="Components from within the application"
					/>
				</div>
			</div>
			<div className="landing-page__section-4">
				<div>
					<img
						src="./images/macroComponent.png"
						alt="Components from within the application"
					/>
				</div>
				<div>
					<h1>Track your macros to optimize your gains</h1>
				</div>
			</div>
			<AppFooter />
		</div>
	);
};

const mapDispatchToProps = (dispatch) => ({
	startLogin: () => dispatch(startLogin),
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
