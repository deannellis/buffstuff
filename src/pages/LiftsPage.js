import React, { Component } from 'react';
import { withRouter } from "react-router-dom";
import { connect } from 'react-redux';
import SideNav from '../components/SideNav';
import Tabs from '../components/Tabs';
import LineGraph from '../components/LineGraph';

class LiftsPage extends Component {
    constructor(props) {
        super(props);
        this.state = {  
            activeTab: 0
        }
    }

    getData = () => {
        const { workouts } = this.props;
        let filteredWorkouts = [];
        workouts.map(workout => {
            let workoutWithDate = {}
            if(workout.hasOwnProperty('squat')) {
                workoutWithDate = {
                    ...workout['squat'],
                    date: workout.created
                }
                filteredWorkouts.push(workoutWithDate)
            }
        })
        // console.log('fw', filteredWorkouts);
        return filteredWorkouts
    }

    render() { 
        return (
            <div className="page--with-side-nav">
                <SideNav path={this.props.match.path} />
                <div className="lifts-page">
                    <Tabs 
                        activeIndex={this.state.activeTab} 
                        handleSelect={i => this.setState({ activeTab: i })}
                        labels={['Bench Press', 'Deadlift', 'Overhead Press', 'Row', 'Squat']}
                    >
                        <LineGraph data={this.getData()}/>

                        {/* {this.state.activeTab === 0 && 
                            <p>1</p>
                        }
                        {this.state.activeTab === 1 && 
                            <p>2</p>
                        }
                        {this.state.activeTab === 2 && 
                            <p>3</p>
                        }
                        {this.state.activeTab === 3 && 
                            <p>4</p>
                        }
                        {this.state.activeTab === 4 && 
                            <p>5</p>
                        } */}
                    </Tabs>
                </div>
            </div>
        );
    }
}
 
const mapStateToProps = state => {
    return {
        workouts: state.workouts,
    };
}
 
export default withRouter(connect(mapStateToProps)(LiftsPage));