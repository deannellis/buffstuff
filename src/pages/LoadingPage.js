import React from 'react';

const LoadingPage = () => {
	return (
		<div className="loading-page">
			<img alt="loading spinner" src="./images/Spinner-1s-200px.gif" />
		</div>
	);
};

export default LoadingPage;
