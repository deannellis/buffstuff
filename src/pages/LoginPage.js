import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

import { startLogin } from '../actions/auth';
import Button from '../components/Button';

const LoginPage = ({ startLogin }) => {
	return (
		<>
			<h1>Welcome to Flext</h1>
			<Button clickHandler={startLogin()}>Login to Flext</Button>
			<button>
				<Link to="/home">To home</Link>
			</button>
		</>
	);
};

const mapDispatchToProps = (dispatch) => ({
	startLogin: () => dispatch(startLogin),
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
