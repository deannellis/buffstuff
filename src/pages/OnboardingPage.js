import React, { Component } from "react";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
import { setMasterWeights } from "../actions/masterWeights";
import OnboardingCard from "../components/OnboardingCard";
import { MenuContext } from "../context/menu-context";

export class OnboardingPage extends Component {
	constructor(props) {
		super(props);
		this.state = {};
	}

	onMasterWeightsSubmit = masterWeights => {
		// this.props.dispatch(setMasterWeights(masterWeights));
		this.props.onMasterWeightsSubmit(masterWeights);
		this.props.history.push("/home");
	};

	componentDidMount() {
		this.context.setPageMenu(false);
		if (
			Object.entries(this.props.masterWeights).length !== 0 &&
			this.props.masterWeights.constructor === Object
		) {
			this.props.history.push("/home");
		}
	}

	componentWillUnmount() {
		this.context.setPageMenu(true);
	}

	render() {
		return (
			<div className="onboarding-page__wrapper">
				<div className="card">
					<OnboardingCard onSubmit={this.onMasterWeightsSubmit} />
				</div>
			</div>
		);
	}
}
OnboardingPage.contextType = MenuContext;
OnboardingPage.defaultProps = {
	masterWeights: {}
};

const mapStateToProps = state => {
	return {
		masterWeights: state.masterWeights
	};
};
const mapDispatchToProps = dispatch => ({
	onMasterWeightsSubmit: masterWeights =>
		dispatch(setMasterWeights(masterWeights))
});

export default withRouter(
	connect(mapStateToProps, mapDispatchToProps)(OnboardingPage)
);
