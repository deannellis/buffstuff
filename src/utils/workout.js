import moment from 'moment';

export const getWorkouts = ({ a = 0, b = 0 }) => {
    let workouts = []
    if(a == 0) {
        workouts.push('overhead', 'chinup')
    } else {
        workouts.push('bench', 'row')
    }
    if(b == 0 || b == 2) {
        workouts.push('squat');
    } else {
        workouts.push('deadlift');
    }
    return workouts;
}

export const getDisplayName = name => {
    switch(name) {
        case 'bench':
            return 'Bench Press';
        case 'row':
            return 'Row';
        case 'squat':
            return 'Squat';
        case 'deadlift':
            return 'Deadlift';
        case 'overhead':
            return 'Overhead Press';
        case 'chinup':
            return 'Chin-Ups';
        default:
            return name;
    }
};

export const getWarmupWeights = workWeight => {
    const warmup25 = workWeight * 0.25 > 44 ? workWeight * 0.25 : 44;
    const warmup50 = workWeight * 0.50 > 44 ? workWeight * 0.50 : 44;
    const warmup75 = workWeight * 0.75 > 44 ? workWeight * 0.75 : 44;
    return [ warmup25, warmup50, warmup75 ];
}

export const getWeightDistribution = weight => {
    if(weight === 44) {
        return [ 0, 44, 0]
    }
    const oneSide = (weight - 44) / 2;
    return [ oneSide, 44, oneSide ];
}

export const getMonthWorkouts = workouts => {
    let dates = {};
    workouts.forEach(workout => {
        let month = moment(workout.created).format("MMMM");
        let day = parseInt(moment(workout.created).format("D"));
        if(dates.hasOwnProperty(month)) {
            dates[month].push(day);
        } else {
            dates[month] = [ day ];
        }
    });
    // console.log(dates);
    return dates;
}

export const getWorkoutDays = (start, daysInMonth) => {
    let workoutDays = [];
    let checkEven = false;
    let weekday = start + 1;
    for(let d = 1; d <= daysInMonth; d++) {
        if(checkEven) {
            if(d%2 == 0) workoutDays.push(d)
        } else {
            if(d%2 !== 0) workoutDays.push(d)
        }
        if(weekday == 7){
            weekday = 0;
            checkEven = !checkEven;
        };
        weekday ++;
    }
    return workoutDays;
};
