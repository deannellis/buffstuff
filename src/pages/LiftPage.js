import React, { Component } from 'react';
import { withRouter } from "react-router";
import { connect } from 'react-redux';
import { updateWorkout } from '../actions/inProgressWorkout';
import { getDisplayName, getWarmupWeights, getWeightDistribution } from '../utils/workout';
import Button from '../components/Button';

class LiftPage extends Component {
    constructor(props) {
        super(props);
        this.state = {  }
    }

    completeLift = (result) => {
        const liftResult = {updates: {[this.props.match.params.id]: result}}
        console.log(liftResult);
        this.props.dispatch(updateWorkout(liftResult));
        this.props.history.push('/workout');
    }

    render() { 
        let { id } = this.props.match.params;
        const workWeight = this.props.masterWeights[id];
        const workReps = ['5 reps', '5 reps', 'AMRAP*'];
        const { negatives, ups, weight } = this.props.masterWeights.chinup;
        return (
            <div className="lift-page">
                <div className="lift-page__card">
                    <h2>
                        {getDisplayName(id)}
                    </h2>
                    {id === 'chinup' ? (
                        <div>
                            <h3>Complete</h3>
                            <p className="lift-page__chinup-title">3 sets of</p>
                            <div className="lift-page__chinups">
                                {negatives === 0 ? '' : (
                                    <div className="lift-page__chinup">
                                        <p>{negatives}</p>
                                        <span>negatives</span>
                                    </div>
                                )}
                                {ups === 0 ? '' : (
                                    <div className="lift-page__chinup">
                                        <p>{ups}</p>
                                        <span>chin-ups</span>
                                    </div>
                                )}
                                {weight === 0 ? '' : (
                                    <div className="lift-page__chinup">
                                        <p>{weight}</p>
                                        <span>lbs</span>
                                    </div>
                                )}
                            </div>
                            <h3>Enter your results</h3>
                            <p>Were you able to complete all sets and reps?</p>
                            <Button variant="primary" clickHandler={() => {this.completeLift(1)}}>Yes</Button>
                            <Button clickHandler={() => {this.completeLift(0)}}>No</Button>
                        </div>
                    ) : (
                        <>
                            <div className="lift-page__tables">
                                <div>
                                    <h3>Warm-Up</h3>
                                    {getWarmupWeights(workWeight).map((weight, i) => (
                                        <div key={i} className="lift-page__sets">
                                            <div className="lift-page__set">
                                                <span>Set {i + 1}</span>
                                                <p>5 reps @{weight}lbs</p>
                                            </div>
                                            <div className="lift-page__distributed">
                                                {getWeightDistribution(weight).map((distWeight, i) => (
                                                    <div key={i} className="lift-page__distributed-item">
                                                        <span>
                                                            {i === 0 && 'LEFT'}
                                                            {i === 1 && 'BAR'}
                                                            {i === 2 && 'RIGHT'}
                                                        </span>
                                                        <p>{distWeight}</p>
                                                    </div>
                                                ))}
                                            </div>
                                        </div>
                                    ))}
                                </div>
                                <div>
                                    <h3>Work</h3>
                                    {workReps.map((rep, i) => (
                                        <div key={i} className="lift-page__sets">
                                                <div className="lift-page__set">
                                                    <span>Set {i + 1}</span>
                                                    <p>{rep} @{workWeight}lbs</p>
                                                </div>
                                                <div className="lift-page__distributed">
                                                    {getWeightDistribution(workWeight).map((distWeight, i) => (
                                                        <div key={i} className="lift-page__distributed-item">
                                                            <span>
                                                                {i === 0 && 'LEFT'}
                                                                {i === 1 && 'BAR'}
                                                                {i === 2 && 'RIGHT'}
                                                            </span>
                                                            <p>{distWeight}</p>
                                                        </div>
                                                    ))}
                                                </div>
                                        </div>
                                    ))}
                                </div>
                            </div>
                            <p>*as many reps as possible </p>
                            <h3 className="lift-page__results-title">Enter your results</h3>
                            <p>On your last set how many reps did you complete?</p>
                            <Button clickHandler={() => {this.completeLift(0)}}>Less than 5</Button>
                            <Button clickHandler={() => {this.completeLift(1)}}>Greater than 5</Button>
                            <Button clickHandler={() => {this.completeLift(2)}}>Greater than 10</Button>
                        </>
                    )}
                    
                </div>
            </div>
        );
    }
}
 
const mapStateToProps = state => {
    return {
        masterWeights: state.masterWeights,
    };
}
 
export default withRouter(connect(mapStateToProps)(LiftPage));