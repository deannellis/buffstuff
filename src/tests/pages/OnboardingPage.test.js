import React from "react";
import PropTypes from "prop-types";
import { shallow } from "enzyme";

import { OnboardingPage } from "../../pages/OnboardingPage";
import { weights, workouts } from "../fixtures/workout";

OnboardingPage.contextTypes = {
	setPageMenu: PropTypes.func
};
const context = { setPageMenu: () => {} };

test("should render the Onboarding Page without props", () => {
	const wrapper = shallow(<OnboardingPage />, { context });
	expect(wrapper).toMatchSnapshot();
});

test("should route user to homepage if masterweights is not an empty object", () => {
	const history = { push: jest.fn() };
	const wrapper = shallow(
		<OnboardingPage masterWeights={weights} history={history} />,
		{ context }
	);

	expect(history.push).toHaveBeenCalledWith("/home");
});

test("should handle onMasterWeightsSubmit", () => {
	const onMasterWeightsSubmit = jest.fn();
	const history = { push: jest.fn() };
	const wrapper = shallow(
		<OnboardingPage
			onMasterWeightsSubmit={onMasterWeightsSubmit}
			history={history}
		/>,
		{ context }
	);

	wrapper.find("OnboardingCard").prop("onSubmit")(weights);

	expect(onMasterWeightsSubmit).toHaveBeenCalledWith(weights);
	expect(history.push).toHaveBeenCalledWith("/home");
});
