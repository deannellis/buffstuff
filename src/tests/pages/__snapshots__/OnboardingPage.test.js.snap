// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Onboarding Page without props 1`] = `
<div
  className="onboarding-page__wrapper"
>
  <div
    className="card"
  >
    <OnboardingCard
      onSubmit={[Function]}
    />
  </div>
</div>
`;
